rpcclient/infrastructure.go

```diff
@@ -815,6 +817,14 @@ func (c *Client) handleSendPostMessage(jReq *jsonRequest) {
 		return
 	}
 
+	// We still want to return an error if for any reason the respone
+	// remains empty.
+	if httpResponse == nil {
```

Comment:
What about capturing that _final_ error to give the caller a more descriptive message? So in this case we'd make a new like` httpErr` error variable, which is used here (and also checked above?).

Reply:
IUUC what you mean, then the last commit I added should address it.